android-core/src/main/java/com/mparticle/internal/ConfigManager.java

```diff
@@ -1222,7 +1207,7 @@ public String getPreviousAdId() {
     public void setPreviousAdId() {
         MPUtility.AdIdInfo adInfo = MPUtility.getAdIdInfo(mContext);
         String currentAdId = null;
-        if (adInfo != null) {
+        if (adInfo != null && !adInfo.isLimitAdTrackingEnabled) {
```

Comment:
similar situation to above - if the ad id is now null - shouldn't be either be removing the PREVIOUS_ANDROID_ID, setting it to null, all zeroes, or something other than keeping the old one?

Reply:
i removed `PREVIOUS_ANDROID_ID` shared pref when LAT is on